app/src/test/java/uk/ac/ebi/atlas/experimentpage/ExperimentControllerWIT.java

```diff
@@ -68,13 +69,18 @@ void cleanDatabaseTables() {
     @MethodSource("publicExperimentsProvider")
     void validExperimentAccession(String experimentAccession) throws Exception {
         mockMvc.perform(get(URL, experimentAccession))
-                    .andExpect(status().isOk())
-                    .andExpect(view().name("experiment-page"))
-                    .andExpect(model().attribute("experimentAccession", experimentAccession))
-                    .andExpect(model().attribute(
-                            "type", ExperimentType.SINGLE_CELL_RNASEQ_MRNA_BASELINE.getHumanDescription()))
-                    .andExpect(model().attributeExists("numberOfCells"))
-                    .andExpect(model().attributeExists("content"));
+                .andExpect(status().isOk())
+                .andExpect(view().name("experiment-page"))
+                .andExpect(model().attribute("experimentAccession", experimentAccession))
+                .andExpect(model().attributeExists("type"))
+                .andExpect(model().attributeExists("numberOfCells"))
+                .andExpect(model().attributeExists("content"));
+
+        var experimentTypeHumanDescription = mockMvc.perform(get(URL, experimentAccession))
+                .andReturn().getModelAndView().getModel().get("type");
+        assertThat(experimentTypeHumanDescription)
+                .isIn(ExperimentType.SINGLE_CELL_RNASEQ_MRNA_BASELINE.getHumanDescription(),
+                        ExperimentType.SINGLE_NUCLEUS_RNASEQ_MRNA_BASELINE.getHumanDescription());
```

Comment:
Good to keep the semantics of the test. It’s a shame that we can’t make more sophisticated assertions from `mockMvc`.